ratis-server/src/main/java/org/apache/ratis/server/impl/LeaderState.java

```diff
@@ -787,6 +789,46 @@ private void replicateNewConf() {
     return lists;
   }
 
+  /**
+   * See the thesis section 6.2: A leader in Raft steps down
+   * if an election timeout elapses without a successful
+   * round of heartbeats to a majority of its cluster.
+   */
+  private void checkLeadership() {
+    // The initial value of lastRpcResponseTime in FollowerInfo is set by
+    // LeaderState::addSenders(), which is fake and used to trigger an
+    // immediate round of AppendEntries request. To ensure that leader
+    // is stable, just ignore the check if become leader soon.
+    if (server.getRole().getRoleElapsedTimeMs() < 3 * server.getMaxTimeoutMs()) {
```

Comment:
Thanks for pointing this. I will remove this magic number 3. 

Basically, candidates collect votes from majority before becoming leader. Since response time of request votes is recoded, If leader just take charge, without seeing higher term, ideally it should be legal for election timeout.